googleapis_auth/lib/src/auth_functions.dart

```diff
@@ -16,6 +16,23 @@ import 'exceptions.dart';
 import 'http_client_base.dart';
 import 'utils.dart';
 
+/// Obtains a HTTP client which uses the given [apiKey] for making HTTP
+/// requests.
+///
+/// Note that the returned client should *only* be used for making HTTP requests
+/// to Google Services. The [apiKey] should not be disclosed to third parties.
+///
+/// The user is responsible for closing the returned HTTP [Client].
+/// Closing the returned [Client] will not close [baseClient].
+Client clientViaApiKey(String apiKey, {Client? baseClient}) {
+  if (baseClient == null) {
+    baseClient = Client();
```

Comment:
This `Client()` is "the right one" for browser vs IO. This could have been done a long time ago!

Reply:
I don't even understand why `BrowserClient` is a publicly visible type.

Reply:
https://pub.dev/documentation/http/latest/browser_client/BrowserClient/withCredentials.html